cartography/intel/azure/network.py

```diff
@@ -5,98 +5,574 @@
+    run_cleanup_job('azure_import_networks_subnets_cleanup.json', neo4j_session, common_job_parameters)
+
+
+def sync_networks_subnets(neo4j_session: neo4j.Session, credentials: Credentials, subscription_id: str, update_tag: int, common_job_parameters: Dict,) -> None:
+    networks_subnets_list = get_networks_subnets_list(credentials, subscription_id)
+    load_networks_subnets(neo4j_session, networks_subnets_list, update_tag)
+    cleanup_networks_subnets(neo4j_session, common_job_parameters)
+
+
+def get_network_routetables_list(credentials: Credentials, subscription_id: str) -> List[Dict]:
+    try:
+        client = get_network_client(credentials, subscription_id)
+        network_routetables_list = list(map(lambda x: x.as_dict(), client.route_tables.list_all()))
+
+        for routetable in network_routetables_list:
+            x = routetable['id'].split('/')
+            routetable['resource_group'] = x[x.index('resourceGroups') + 1]
+        return network_routetables_list
+
+    except HttpResponseError as e:
+        logger.warning(f"Error while retrieving routetables - {e}")
+        return []
+
+
+def _load_network_routetables_tx(tx: neo4j.Transaction, subscription_id: str, network_routetables_list: List[Dict], update_tag: int) -> None:
+    ingest_routetables = """
+    UNWIND {network_routetables_list} AS routetable
+    MERGE (n:AzureRoutetable{id: routetable.id})
+    ON CREATE SET n.firstseen = timestamp(),
+    n.type = routetable.type,
+    n.location = routetable.location,
+    n.resourcegroup = routetable.resource_group
+    SET n.lastupdated = {update_tag},
+    n.name = routetable.name,
+    n.etag=routetable.etag
+    WITH n
+    MATCH (owner:AzureSubscription{id: {SUBSCRIPTION_ID}})
+    MERGE (owner)-[r:RESOURCE]->(n)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {update_tag}
+    """
+
+    tx.run(
+        ingest_routetables,
+        network_routetables_list=network_routetables_list,
+        SUBSCRIPTION_ID=subscription_id,
+        update_tag=update_tag,
+    )
+
+
+def cleanup_network_routetables(neo4j_session: neo4j.Session, common_job_parameters: Dict) -> None:
+    run_cleanup_job('azure_import_network_routetables_cleanup.json', neo4j_session, common_job_parameters)
+
+
+def sync_network_routetables(
+    neo4j_session: neo4j.Session, credentials: Credentials, subscription_id: str, update_tag: int,
+    common_job_parameters: Dict,
+) -> None:
+    networks_list = get_network_routetables_list(credentials, subscription_id)
+    load_network_routetables(neo4j_session, subscription_id, networks_list, update_tag)
+    cleanup_network_routetables(neo4j_session, common_job_parameters)
+
+
+@ timeit
+def get_network_routes_list(credentials: Credentials, subscription_id: str) -> List[Dict]:
+    try:
+        client = get_network_client(credentials, subscription_id)
+        network_routetables_list = get_network_routetables_list(credentials, subscription_id)
```

Comment:
we should not fetch route tables again. Instead, we should pass route tables to this function `get_network_routes_list `. and iterate over route tables to fetch routes.